### PR TITLE
Aditya 25 nwsensor

### PR DESCRIPTION
**Pull request by adityaguptakd**

### CODE REVIEW

File: ceaos/actuators/actuator_definition.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    def is_running(self):
+        return self.running
+
+    def calibrate(self):
+        """
+        This method is used to calibrate the actuator.
+        Need not be implemented for actuators that don't require calibration
+        """
+        pass
```
Comment (nandantumu) [inline ceaos/actuators/actuator_definition.py:26]: Newline

File: ceaos/actuators/sample_actuator.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    def calibrate(self, calib_val):
+        pass
+
+    # returns the measurement the actuator is actuating (i.e. temperature, pH)
+    def get_datatype(self):
+        return self.datatype
+
+    def set_datatype(self, datatype):
+        self.datatype = datatype
```
Comment (nandantumu) [inline ceaos/actuators/sample_actuator.py:42]: Newline

File: ceaos/actuators/sample_actuator.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
+from .actuator_definition import Actuator
+from ..sensors.sensor_definition import Sensor
+
+
+class Artificial_Actuator(Actuator):
```
Comment (nandantumu) [inline ceaos/actuators/sample_actuator.py:5]: Can you change the class name to `ArtificialActuator`?

File: docker-compose.yml [hunk 4]
```diff
@@ -3,6 +3,10 @@ services:
     cea-os:
         build: . #builds image from Dockerfile in current directory
         container_name: cea_os
+        ports:
+            - 23267:23267
+        expose:
+            - 23267
```
Comment (nandantumu) [inline docker-compose.yml:9]: You should only need one of these

File: ceaos/sensors/sensor_definition.py [hunk 5]
```diff
@@ -12,9 +12,3 @@ def read_value(self):
         This method returns the value of the sensor
         """
         raise NotImplementedError
-
-    def calibrate(self):
-        """
-        This method is used to calibrate the sensor. Need not be implemented for sensors that don't require calibration
-        """
-        pass
```
Comment (nandantumu) [inline ceaos/sensors/sensor_definition.py:20]: Why did you delete this?

File: scripts/runner.py [hunk 6]
```diff
@@ -49,37 +49,48 @@ def log_data(refresh_rate, logger_list, client, farm_name):
...
     for refresh_rate, logger_list in loggers.items():
         thread = threading.Thread(
-            target=log_data, args=(refresh_rate, logger_list, db_client, farm_name)
+            target=log_data,
+            args=(refresh_rate, logger_list, db_client),
+            daemon=True,
         )
         threads.append(thread)
-
+    print("Threads: ", len(threads))
```
Comment (nandantumu) [inline scripts/runner.py:80]: Can you change this print statement to a logging statement? Perhaps merge with the statement below

**Review by nandantumu — CHANGES_REQUESTED** [review]

**Comment by adityaguptakd** [comment]
Also Nandan, in the main branch we no longer need stuff like temp_sens.py that was originally comitted because we didn't have the network sensor idea at the time, but now that stuff is completely useless. Not sure if there is a way for you to decommit it

**Comment by nandantumu** [comment]
Yup, we can just make a refactoring commit that deletes those.

**Review by nandantumu — APPROVED** [review]
LGTM!

**Pull request merged**